src/Nethermind/Nethermind.JsonRpc/Modules/Eth/GasPrice/GasPriceOracle.cs

```diff
@@ -148,6 +153,7 @@ private IEnumerable<UInt256> GetGasPricesFromRecentBlocks(IEnumerable<Block> blo
 
                 if (txFromCurrentBlock == 0)
                 {
+                    blocksToGoBack--;
```

Comment:
Did we have any reason for not decreasing it for empty blocks? In gnosis devnet when we have 100% of blocks empty we're iterating through entire BlockTree over and over https://github.com/NethermindEth/nethermind/blob/master/src/Nethermind/Nethermind.JsonRpc/Modules/Eth/GasPrice/GasPriceOracle.cs#L111

Reply:
The idea was to try calculate gas price from real transactions instead of empty blocks, but yeah if most are empty then its a problem